prov/efa/src/efa_domain.c

```diff
@@ -309,8 +324,23 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 	if (app_mr_local || !efa_mr_cache_enable)
 		return 0;
 
-	if (efa_check_fork_enabled(*domain_fid))
-		efa_fork_status = EFA_FORK_SUPPORT_ON;
+	/*
+	 * Call ibv_fork_init if the user asked for fork support, otherwise
+	 * check if it was enabled by something else. Huge pages for bounce
+	 * buffers will not be used if fork support is on.
+	 */
+	if (efa_fork_status == EFA_FORK_SUPPORT_ON) {
+		ret = -ibv_fork_init();
```

Comment:
I think call ibv_fork_init here is too late, in `efa_check_fork_enabled` we registered a pointer. If ibv_fork_init has not been called prior to that, this call would always fail.

Reply:
I'm not sure I understand the concern as `efa_check_fork_enabled` is called after this. But I don't see any harm in calling `ibv_fork_init` in `rxr_init_env` and can move it there.

Reply:
Sorry, I misread the change. (I did not notice that efa_fork_fork_enabled is to be moved).

But I would prefer ibv_fork_init to be called earlier. Thanks!